.github/workflows/test_and_make_release.yml

```diff
@@ -12,24 +12,131 @@ on:
       - master
 
 jobs:
-  test:
+  template_test:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.8]
+        python-version: [3.8, 3.7, 3.6]
     steps:
       - uses: actions/checkout@v2
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v1
         with:
           python-version: ${{ matrix.python-version }}
+      - name: Install tools (Poetry)
+        uses: Gr1N/setup-poetry@v2
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
+          pip install tox
+          pip install cookiecutter
+      - name: Instantiate template and validate
+        env:
+          POETRY_VIRTUALENVS_CREATE: false
+        run: |
+          cookiecutter --default-config --no-input .
+          cd python_boilerplate
+          make test
+
+  template_bdd:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.6
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.6
+      - name: Install tools (Poetry)
+        uses: Gr1N/setup-poetry@v2
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install tox
+          pip install cookiecutter
+      - name: Run bdd tests
+        env:
+          POETRY_VIRTUALENVS_CREATE: false
+        run: |
+          cookiecutter --default-config --no-input .
+          cd python_boilerplate
+          make bdd
+
+  template_format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python 3.8
```

Comment:
Why no matrix strategy like previous job ?

Reply:
It is formatting of the code, so only needs a python interpreter to run and do its checks.